File: Java/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int min = Integer.MAX_VALUE;
        int[] answer = new int[2];
        
        int R = 0;
        int L = 0;
        int sum = 0;
        for (; L<sequence.length; L++) {
            while (R<sequence.length && sum < k) {
                sum += sequence[R++];
            }
            
            if (sum == k && min > R-L) {
                    answer[0] = L;
                    answer[1] = R-1;
                    min = R-L;
            }
            
            sum -= sequence[L];
        }
        
        
        return answer;
    }
}

File: Swift/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.swift
import Foundation

func solution(_ sequence:[Int], _ k:Int) -> [Int] {
    var pointerOne: Int = 0
    var pointerTwo: Int = 0
    var total: Int = 0
    var answer: [Int] = [0, 0]
    var min = Int.max
    while pointerOne < sequence.count {
        while pointerTwo < sequence.count && total < k {
            total += sequence[pointerTwo]
            pointerTwo += 1
        }
        
        if total == k && min > (pointerTwo - pointerOne + 1) {
            answer[0] = pointerOne
            answer[1] = pointerTwo - 1
            min = pointerTwo - pointerOne + 1
        } 
        
        total -= sequence[pointerOne]
        pointerOne += 1
    }
    
    return answer
}